crypto/threads_pthread.c

```diff
@@ -24,6 +24,10 @@
 #  include <unistd.h>
 #endif
 
+# if defined(OPENSSL_SYS_TANDEM)
+#  include <assert.h>
+# endif
```

Comment:
IMO this should be included not just on OPENSSL_SYS_TANDEM but everywhere. Actually it is not needed if USE_RWLOCK is defined below, but IMO it is harmless to always include it.

Reply:
That's what I thought, but apparently the CI builds don't seem to need it. Unless the implicit definition is not noticed. So remove the #if/#endif?

Reply:
Agreed

Reply:
>  apparently the CI builds don't seem to need it

Most probably because the USE_RWLOCK is being defined for them.